### ISSUE TITLE
[Android] EdgeToEdge does not work for Modal page

### ISSUE THREAD

**Issue by tranb3r** [dotnet/maui#24903]
### Description

When EdgeToEdge is enabled on Android, a Modal page is not filling the whole screen, it is starting below the system status bar. The system status bar is visible and the bottom of the Modal page is invisible.

On my example: 
- the mainpage (pink background) is displayed edge-to-edge, while the modal page (blue background) is displayed below the system status bar.
- the text in the modal page does not seem to be vertically centered, because the modal page is not occupying the whole screen

![test](https://github.com/user-attachments/assets/bfc7a478-f037-4b09-b4bc-797f6d040041)

This happens both with the EdgeToEdge extension and when manually setting edge-to-edge (see code in MainActivity).

### Steps to Reproduce

1. Open the repro project and run on android
2. Click the button to open a modal page
3. Click on back button to close the modal page

### Link to public reproduction project repository

https://github.com/tranb3r/Issues/tree/main/MauiAppEdgeToEdge

### Version with bug

8.0.90 SR9

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 34

### Did you find any workaround?

No

**Comment by drasticactions** [comment]
I don't think EdgeToEdge is relevant for what you're trying to do here. That's not what the bug is. I think it's https://github.com/dotnet/maui/issues/18033

The top bar insets are still there, and the model view is going up to those insets. If you step through the ModelNavigationManager code here, you would hit those margins

https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs#L330-L343

There was a change for allowing it to go "Edge to Edge" with (https://github.com/dotnet/maui/pull/19903) but you need to enable "Immersive Mode" within your application (Like [this code](https://github.com/ewerspej/dotnet-maui-bug-repro-modal-fullscreen/blob/main/bug-modal-fullscreen/bug-modal-fullscreen/Platforms/Android/DeviceServices.cs) does), it will go "Edge to Edge"

But that's not solving the core issue of the top margin bar on the system bar. "EdgeToEdge" doesn't change that. So I think this a duplicate of https://github.com/dotnet/maui/issues/18033 and that needs to be fixed to figure out how to really handle the status bar margin.

**Comment by jaosnz-rep** [comment]
I can repro this issue at Android platform on the latest 17.12.0 Preview 2.0(8.0.82 & 8.0.90).

**Comment by Redth** [comment]
This is slightly different on .NET 9 RC1 given @pictos changes to the android modal page using dialogs:


https://github.com/user-attachments/assets/ec26778f-66d6-4d52-a267-8a878d502ea0

I see we now do not set inset padding, but we are using a style: https://github.com/dotnet/maui/blob/net9.0/src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs#L326

Perhaps we need another style for dialogs that defines some more things to avoid the opaque status bar.

**Comment by tranb3r** [comment]
@Redth 
I've tested the solution you've proposed.
I can confirm it fixes the modal issue at the top of the screen.
However, using net9-rc2, there is now an issue at the bottom of the screen: the system navigation bar is appearing solid-white, both for the main page and for the modal page, instead of having the same color as the page (when using edge-to-edge). Is this expected? Can this be fixed?
I've tried to set navigationbar color using maui community toolkit, it does work for the main page, but not for the modal page.

**Comment by borrmann** [comment]
Since .NET 9 modal pages also break other dialogs on android. Popups from the MAUI Community Toolkit or Bottom Sheets from The49 are now displayed behind modal pages.

**Comment by Elinareso** [comment]
> Since .NET 9 modal pages also break other dialogs on android. Popups from the MAUI Community Toolkit or Bottom Sheets from The49 are now displayed behind modal pages.

Yes, also a third party controls like Telerik Popup, List, Date, etc. The SnackBar is being displayed behind of first page.

https://feedback.telerik.com/maui/1673812-pickers-android-do-not-display-in-a-shell-modal-page?_gl=1*127cnzr*_gcl_au*OTExMDM2NDExLjE3MzQzNjczNzI.*_ga*MTkzODAxMTI2My4xNjQ1NTQ2ODcx*_ga_9JSNBCSF54*MTczNTMzMTcwMi40LjEuMTczNTMzMzgwMi42MC4wLjA.

![Image](https://github.com/user-attachments/assets/883ff4cd-2cd8-4e29-a3c0-55a8af3b5e38)

**Comment by pictos** [comment]
@elinareso @borrmann do you have a small sample project that reproduces the issue?

**Comment by Elinareso** [comment]
> [@Elinareso](https://github.com/Elinareso) [@borrmann](https://github.com/borrmann) do you have a small sample project that reproduces the issue?

Yes, I´m attaching an example to reproduce this issue. This example uses the Snackbar, however, this affects several controls, all Telerik controls with popups related.![Image](https://github.com/user-attachments/assets/d1881ef0-ec9a-4c6f-a151-caf42995a74b)

[Issue24903.zip](https://github.com/user-attachments/files/18275335/Issue24903.zip)

**Comment by Elinareso** [comment]
> > [@Elinareso](https://github.com/Elinareso) [@borrmann](https://github.com/borrmann) do you have a small sample project that reproduces the issue?
> 
> Yes, I´m attaching an example to reproduce this issue. This example uses the Snackbar, however, this affects several controls, all Telerik controls with popups related.
> 
> ![Image](https://github.com/user-attachments/assets/d1881ef0-ec9a-4c6f-a151-caf42995a74b) [ ![Image](https://github.com/user-attachments/assets/d1881ef0-ec9a-4c6f-a151-caf42995a74b) ](https://private-user-images.githubusercontent.com/18618697/399309860-d1881ef0-ec9a-4c6f-a151-caf42995a74b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1NjkwMTgsIm5iZiI6MTczNTU2ODcxOCwicGF0aCI6Ii8xODYxODY5Ny8zOTkzMDk4NjAtZDE4ODFlZjAtZWM5YS00YzZmLWExNTEtY2FmNDI5OTVhNzRiLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDE0MjUxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxOTM4OTg3MTE0NGVkNTExMzI3ZDQwNGJiMjhmNTU2M2E0OWFhMjk3ZTIwZGQ4NmFiN2ViOTBkYjdlYzcyMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9cO0ufJ4R-shG3BwEI-gTexp7raC_glcBtuRH2YWBoY) [ ](https://private-user-images.githubusercontent.com/18618697/399309860-d1881ef0-ec9a-4c6f-a151-caf42995a74b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1NjkwMTgsIm5iZiI6MTczNTU2ODcxOCwicGF0aCI6Ii8xODYxODY5Ny8zOTkzMDk4NjAtZDE4ODFlZjAtZWM5YS00YzZmLWExNTEtY2FmNDI5OTVhNzRiLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDE0MjUxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxOTM4OTg3MTE0NGVkNTExMzI3ZDQwNGJiMjhmNTU2M2E0OWFhMjk3ZTIwZGQ4NmFiN2ViOTBkYjdlYzcyMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9cO0ufJ4R-shG3BwEI-gTexp7raC_glcBtuRH2YWBoY)
> 
> [Issue24903.zip](https://github.com/user-attachments/files/18275335/Issue24903.zip)

![Image](https://github.com/user-attachments/assets/7ef66352-4ea5-42b1-93db-2db7ba6b2d14)
[Issue24903.zip](https://github.com/user-attachments/files/18275627/Issue24903.zip)

**Comment by pictos** [comment]
@Elinareso thanks, I'll take a look on it

**Comment by pictos** [comment]
@Elinareso actually, you are using a wrong versoin of MCT, the version that you used on your sample (9.X.X) doesn't have support for .net 9 maui, if I bump it to 10.0.0 it works as expected.

![Image](https://github.com/user-attachments/assets/386cff81-c9df-40ab-8746-832833d2f714)

**Comment by Elinareso** [comment]
> [@Elinareso](https://github.com/Elinareso) actually, you are using a wrong versoin of MCT, the version that you used on your sample (9.X.X) doesn't have support for .net 9 maui, if I bump it to 10.0.0 it works as expected.
> 
> ![Image](https://github.com/user-attachments/assets/386cff81-c9df-40ab-8746-832833d2f714)

Thanks so much ! Yes, I updated the nuget and this control is working like a charm, however, the Telerik controls are not, I reported this to the Telerik team.

Happy New Year! Party popper

**Comment by callidusviro** [comment]
@Redth I am currently trying to get modals working with Edge to Edge on android, and have tried following this approach, though I am unable to use the EnableEdgeToEdge method as it seemingly doesn't exist. I am currently using .NET 9. Is there something else I need to be adding to access the EnableEdgeToEdge method?
